spec/logr_spec.rb

```diff
@@ -0,0 +1,179 @@
+require 'spec_helper'
+require 'logger'
+require 'json'
+
+describe Logr do
+  context 'default DEBUG log level' do
+    let(:test_output) { StringIO.new }
+    let(:logger) { Logr::Logger.new('test_logger', Logger::DEBUG, test_output) }
+
+    describe '#debug' do
+      before do
+        logger.debug('Test message')
+      end
+
+      it 'should print a timestamp' do
+        logline = JSON.parse(test_output.string)
+        expect(logline.has_key?('timestamp')).to eq(true)
+      end
+
+      it 'should print everything else in the correct structure' do
+        logline = JSON.parse(test_output.string)
+        logline.delete('timestamp')
+        expected_logline = {
+          'logger'  => 'test_logger',
+          'level'   => 'DEBUG',
+          'message' => 'Test message'
+        }
+        expect(logline).to eq(expected_logline)
+      end
+    end
```

Comment:
To test `timestamp` and freeze time, timecop and friends (https://www.ruby-toolbox.com/categories/time_warping) are great. I'm sure this gem will have to deal with time so I would bring it in as a dependency and create better tests.

L22 is a code smell, it should not be necessary. The whole separation should not be necessary.